File: C++/valid_palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        vector<char> alnum;

        for (char c : s){
            if (isdigit(c)){
                alnum.push_back(c);
            }

            else if (isalpha(c)){
                alnum.push_back(tolower(c));
            }
        }

        int l = 0;
        int r = alnum.size() - 1;

        while (l < r){
            if (alnum[l] != alnum[r]){
                return false;
            }
            l++;
            r--;
        }
        return true;
    }
};

File: python/valid_palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        alnum = []

        for c in s:
            if c.isalnum():
                alnum.append(c.lower())

        l, r = 0, len(alnum) - 1
        while l < r:
            if alnum[l] != alnum[r]:
                return False
            l += 1
            r -= 1
        return True
